rkt/app_rm.go

```diff
@@ -61,20 +61,18 @@ func runAppRm(cmd *cobra.Command, args []string) (exit int) {
 		stderr.PrintE("invalid app name", err)
 	}
 
-	if p.State() != pkgPod.Running {
-		stderr.Printf("pod %q is not running", p.UUID)
-		return 254
-	}
-
-	if !p.IsSupervisorReady() {
-		stderr.Printf("supervisor for pod %q is not ready yet", p.UUID)
-		return 254
-	}
-
-	podPID, err := p.ContainerPid1()
-	if err != nil {
-		stderr.PrintE(fmt.Sprintf("unable to determine the pid for pod %q", p.UUID), err)
-		return 254
+	var podPID int
+	if p.State() == pkgPod.Running {
```

Comment:
I don't think you need this. If you are removing an app from a stopped pod, it means it previously reached both pid-file writing and supervisor-ready file writing.

Reply:
I'm using `podPID == 0`  to mean the pod is not running and hence we don't need to call the `stop` and `rm` entrypoints. I guess I was confused by the comment in https://github.com/rkt/rkt/blob/5e83d91aafef5f7a38fef62c045e8b57eeeb8bce/stage0/app_rm.go#L65

Reply:
Which is actually partially true if we use `ppid` because we'll then try to find the `pid` by calling `getChildPID` which looks in `/proc/PPID/task/PPID/children` or over `/proc`. So it will return a `-1`.

However, when we use `pid` (KVM flavor), the file will be present and have a `> 0` value.

Reply:
I could move this test to `stage0.RmApp` to decide whether to call the entrypoints or not.

Reply:
We have a `p.isRunning()` inside `p.ContainerPid1()`. I think I rather leave this like it is now.

Reply:
Ack, I'm fine with that, and thanks for the additional info.